File: python/uav_simulator/capabilities/capability_data.py
class CapabilityData:
    def __init__(self, descriptor: str, capability_bytes):
        """data produced by all capabilities"""
        self.descriptor: str = descriptor
        self.capability_bytes: bytes = capability_bytes

    def __str__(self):
        return f'(Capability: Data{self.descriptor} length:{len(self.capability_bytes)})'

File: cs/uav_ground_control_solution/Calabilities/CapabilityData.cs
namespace Capabilities
{
    public class CapabilityData
    {
        public string Descriptor { get; private set; }
        public byte[] CapabilityBytes { get; private set; }

        // Constructor
        public CapabilityData(string descriptor, byte[] capabilityBytes)
        {
            Descriptor = descriptor;
            CapabilityBytes = capabilityBytes;
        }

        // Override the ToString method for a readable representation
        public override string ToString()
        {
            return $"(Capability: Data{Descriptor} length:{CapabilityBytes.Length})";
        }
    }
}
